### PR TITLE
Refactored to fix 3 implementation smells

### PR DESCRIPTION
**Pull request by ashishbhasin45**
1. Decompose complex conditional.
Location- Sub-project: imixs-workflow-core
Package: org.imixs.workflow.xml
class: XMLItem
Method: stripNonValidXMLCharacters

2. Renamed the variable in the specified method from o to a currentReportItemValue for better understandability.
Location-	Sub-project: imixs-workflow-engine
Package: org.imixs.workflow.engine
Class: ReportService
Method: updateReport

3. Decomposed parseTag method to resolve a complex method code smell for readability and understandability.
Location-	Sub-project: imixs-workflow-core
Package: org.imixs.workflow.util
class: XMLParser
Method: parseTag

### CODE REVIEW

File: imixs-workflow-core/src/main/java/org/imixs/workflow/util/XMLParser.java [hunk 1]
```diff
@@ -396,6 +378,28 @@ public static ItemCollection parseTag(String xmlContent, String tag) throws Plug
         return result;
     }
```
Comment (rsoika) [inline imixs-workflow-core/src/main/java/org/imixs/workflow/util/XMLParser.java:380]: Can you please add the following method comment
and remove the out-comment old code parts. 

/** 
 * This helper method parses all child elements and creates / or appends the tag value as a Item value into the result ItemCollection. 
 * The method is called from parseTag.
 **/

**Review by rsoika — COMMENTED** [review]
Thanks for your contribution!
Yes it was not the the best code ;-)
As we this part is covered by a lot of unit tests we can submit it. Please only add my method comment.

File: imixs-workflow-core/src/main/java/org/imixs/workflow/util/XMLParser.java (hunk 1 above)
Comment (ashishbhasin45) [reply imixs-workflow-core/src/main/java/org/imixs/workflow/util/XMLParser.java:380]: Thanks for the review, added the mentioned comment and removed commented code from the helper method.

Comment (rsoika) [reply imixs-workflow-core/src/main/java/org/imixs/workflow/util/XMLParser.java:380]: Great!

**Pull request merged**